src/lib/core/CHIPConfig.h

```diff
@@ -2797,6 +2797,26 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #ifndef CHIP_CONFIG_ENABLE_SERVER_IM_EVENT
 #define CHIP_CONFIG_ENABLE_SERVER_IM_EVENT 1
 #endif
+
+/**
+ * @def CHIP_CONFIG_MAX_ATTRIBUTE_STORE_ELEMENT_SIZE
+ *
+ * @brief Safety limit to ensure that we don't end up with a
+ * larger-than-expected buffer for temporary attribute storage (on the stack or
+ * in .bss).  The SDK will fail to compile if this value is set below the value
+ * it thinks it needs for a buffer size that can store any simple (not list or
+ * struct) attribute value.
+ */
+#ifndef CHIP_CONFIG_MAX_ATTRIBUTE_STORE_ELEMENT_SIZE
+// Basic Information has a ProductURL attribute that we seem to have enabled
+// everywhere, which is 256 octets.  That leads to a 259-octet buffer size.
+//
+// But I can't figure out how to get all-clusters-app to sanely use a different
+// value here, and that app includes TestCluster, which has very large string
+// attributes (1000 octets, leading to a 1003 octet buffer).
+#define CHIP_CONFIG_MAX_ATTRIBUTE_STORE_ELEMENT_SIZE 1003
```

Comment:
It's a default. If you make your own product (which it would be a grave mistake to copy all-clusters-app, you at least get a chance to override this).

Reply:
Yes, absolutely.  I was just hoping to have the default be a slightly sharper bound.